core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -203,6 +205,30 @@ public Result visit(Project e) {
     return builder.result();
   }
 
+  /** @see #dispatch */
+  public Result visit(Window e) {
+    Result x = visitChild(0, e.getInput());
+    Builder builder = x.builder(e);
+    RelNode input = e.getInput();
+    int inputFieldCount = input.getRowType().getFieldCount();
+    List<SqlNode> rexOvers = new ArrayList<>();
+    for (Window.Group group: e.groups) {
+      rexOvers.addAll(builder.context.toSql(group, e.constants, inputFieldCount));
+    }
+    List<SqlNode> selectList = new ArrayList<>();
+
```

Comment:
Add final modifier for all the containers.

Reply:
ok, I addressed it